src/one_dragon/base/operation/one_dragon_app.py

```
@@ -105,16 +109,28 @@ def switch_instance(self) -> OperationRoundResult:
     @node_from(from_name='切换实例配置', status='游戏路径不同')
     @operation_node(name='关闭游戏', screenshot_before_round=False)
     def close_game(self) -> OperationRoundResult:
-        # 刷新窗口句柄, 避免旧缓存导致误判
-        if self._last_controller is None:
+        # 切换到不同路径的实例时，ctx.switch_instance() 已将 controller 的窗口标题
+        # 更新为新服务器的标题，_last_controller 无法再找到旧游戏窗口。
+        # 改用可执行文件名通过 taskkill 强制关闭旧进程。
+        if self._last_game_path:
+            exe_name = os.path.basename(self._last_game_path)
+            result = subprocess.run(
```

Ruff (0.15.12)

[error] `subprocess` call: check for execution of untrusted input

(S603)